src/components/app-header.tsx

```diff
@@ -1,41 +1,101 @@
+import { Button, ButtonGroup } from "@blueprintjs/core";
 import { inject, observer } from "mobx-react";
 import * as React from "react";
-import { BaseComponent, IBaseProps } from "../../components/base";
-import { GroupModelType, GroupUserModelType } from "../../models/stores/groups";
+import { BaseComponent, IBaseProps } from "./base";
+import { GroupModelType, GroupUserModelType } from "../models/stores/groups";
 
-import "./clue-header.sass";
+import "./app-header.sass";
+
+export interface IPanelSpec {
+  panelId: string;
+  label: string;
+  content: JSX.Element;
+}
+export type IPanelGroupSpec = IPanelSpec[];
 
 interface IProps extends IBaseProps {
   isGhostUser: boolean;
+  panels: IPanelGroupSpec;
+  current: string;
+  onPanelChange: (panel: string) => void;
+  showGroup: boolean;
 }
 
 @inject("stores")
 @observer
-export class ClueHeaderComponent extends BaseComponent<IProps, {}> {
+export class AppHeaderComponent extends BaseComponent<IProps, {}> {
 
   public render() {
-    const {appMode, db, user, problem, groups} = this.stores;
-    const myGroup = groups.groupForUser(user.id);
+    const { showGroup } = this.props;
+    const {appMode, appVersion, db, user, problem, groups} = this.stores;
+    const myGroup = showGroup ? groups.groupForUser(user.id) : undefined;
     const userTitle = appMode !== "authed" ? `Firebase UID: ${db.firebase.userId}` : undefined;
 
     return (
-      <div className="clue-header">
-        <div className="info">
+      <div className="app-header">
+        <div className="left">
           <div>
             <div className="problem" data-test="problem-title">{problem.fullTitle}</div>
             <div className="class" data-test="user-class">{user.className}</div>
           </div>
         </div>
-        {myGroup ? this.renderGroup(myGroup) : null}
-        <div className="user">
-          <div className="name" title={userTitle} data-test="user-name">{user.name}</div>
+        <div className="middle">
+          {this.renderPanelButtons()}
+        </div>
+        <div className="right">
```

Comment:
Do we need to worry about cases where we want to hide the left, middle, or right sections of the header?  I know Sherry expressed wanting some of this hidden in Dataflow at one of the last planning meetings.  Probably best to leave in for now, but we might eventually need to hide one or more of these divs.

Reply:
Eventually, yes, but for now we're in the CLUE/master branch and those issues are not relevant for the immediate story. Now that the `AppHeaderComponent` is common to both CLUE and Dataflow, the expectation is that any differences between the two can be handled with configuration.